framework/materials/multi_group_xs/multi_group_xs.h

```diff
@@ -169,6 +176,24 @@ class MultiGroupXS : public MaterialProperty
   void ComputeDiffusionParameters();
 
   void TransposeTransferAndProduction();
+
+  /// Check vector for all non-negative values
+  bool IsNonNegative(const std::vector<double>& vec)
+  {
+    return not std::any_of(vec.begin(), vec.end(), [](double x) { return x < 0.0; });
+  };
+
+  /// Check vector for all strictly positive values
+  bool IsPositive(const std::vector<double>& vec)
+  {
+    return not std::any_of(vec.begin(), vec.end(), [](double x) { return x <= 0.0; });
+  };
+
+  /// Check vector for any non-zero values
+  bool HasNonZero(const std::vector<double>& vec)
+  {
+    return std::any_of(vec.begin(), vec.end(), [](double x) { return x > 0.0; });
+  };
```

Comment:
I do not understand the comment and the return. If I were to check for non-zero values, I wouldn't do a test on x > 0 . I would check is abs(x) > 0 for example or even abs(x)> tiny

Reply:
I think I would condense these three methods into the following:
```
bool IsGreatherThanZero(...)
{
  return std:all_of(vec.begin(), vec.end(), [](double x) { return x > 0.0; }
}

bool IsGreaterThanOrEqualZero(...)
{
  return std::all_of(vec.begin(), vec.end(), [](double x) { return x >= 0.0; }
}
```

Reply:
I'll submit a new PR with these changes so @zhardy-lanl  can weigh in. That way this on doesn't get held up.

Reply:
@wdhawkins ok, please do a new PR for those checks. I'll approve this one now